commands/promote.js

```diff
@@ -61,6 +63,45 @@ function * run (context, heroku) {
       }
     })
   }))
+
+  let releasePhase = (yield heroku.get(`/apps/${app}/formation`))
+    .find((formation) => formation.type === 'release')
+
+  if (releasePhase) {
+    yield cli.action('Checking release phase', co(function * () {
+      let releases = (yield heroku.get(`/apps/${app}/releases`))
```

Comment:
I'm not sure if this is workable. Shogun (admittedly an extreme case) has 4500+ releases, and paginating through them all (as this will do, 200 at a time) takes over 30s here... I don't know, maybe that's good enough given the alternatives? Or can we assume that the first page of results is almost certain to contain the info we need and stop there? I'm not sure how to opt out of pagination other than [doing everything manually like the api plugin](https://github.com/heroku/heroku-api-plugin/blob/master/src/commands/api.js#L57)--do you folks know off-hand @heroku/cli ?

Reply:
> Or can we assume that the first page of results is almost certain to contain the info we need and stop there?

I believe that the first result definitely should have what we want. I think we can take the same approach of `heroku releases` (not sure where the code is, do you know Maciek?), if I do `heroku releases -n 5`, it only fetches the last 5 releases and that should be pretty enough for us here.

Reply:
Great call @keiko713 -- that code is here: https://github.com/heroku/heroku-apps/blob/master/src/commands/releases/index.js#L22-L30

Reply:
nice! updated.

Reply:
yes, partial: true is what you want